### PR TITLE
fix: add format for timestamp in crdb to correctly use times values

### LINKED ISSUE #15825: [db_specific] CockroachDB does not support to_timestamp
### Expected results

Graph should appear

### Actual results
Error is shown:

> CockroachDB Error
>
> unknown function: to_timestamp() 

#### Screenshots

Without timefilter:

image

With timefilter:

image

#### How to reproduce the bug

Create a graph on database "CockroachDB" and add a time filter

### Environment

(please complete the following information):

- superset version: 1.2
- python version: 3.8
- node.js version: `node -v`
- CockroachDB 21.1.6

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

### Additional context

In Superset 1.0 CockroachDB works fine with time filters. After upgrading to 1.2 this issue appears

### PR DESCRIPTION
**Pull request by elewis787**
### SUMMARY
This PR fixes #15825 

### TESTING INSTRUCTIONS
To test this, the following steps can be taken
- Add db connection for cockroach db. 
- Create a dataset that includes a table with a timestamp or timestampz column, this will be used as the temporal value when creating charts. 
- Create a chart that leverages a temporal value
- Inspect the generated query to see if the date format matches as expected. 

This has been tested in master and version 2.1.0 with python 3.11. 

In version 2.1.0 there were code changes needed to the `dict` type, this is due to a python version that was being used in the docker images of version 2.1.0. 

### ADDITIONAL INFORMATION
- [ x] Has associated issue: #15825

### CODE REVIEW

**Comment by john-bodley** [comment]
Thanks @elewis787 for the PR. Would you mind adding some unit tests? You can look at other DB engine specs for an example of how to do this.

**Comment by elewis787** [comment]
sure - and I will update the link errors.

**Comment by elewis787** [comment]
tests added - I need to run a lint check.

**Comment by elewis787** [comment]
test added and fixed lint error for the db spec file.

**Review by eschutho — APPROVED** [review]
Looks great, thank you @elewis787!

**Comment by elewis787** [comment]
> Looks great, thank you @elewis787!

Thanks @eschutho - I pushed a few updates to hopefully fix the few linit errors that came up.

File: tests/unit_tests/db_engine_specs/test_crdb.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+@pytest.mark.parametrize(
+    "target_type,expected_result",
+    [
+        ("Date", "'2019-01-02'"),
+        ("TimeStamp", "'2019-01-02 03:04:05'"),
+        ("UnknownType", None),
+    ],
+)
+def test_convert_dttm(
+    target_type: str, expected_result: Optional[str], dttm: datetime
```
Comment (john-bodley) [inline tests/unit_tests/db_engine_specs/test_crdb.py:36]:
```suggestion
    target_type: str, expected_result: Optional[str]
```

No need to pass `dttm` as it's in the global scope.

Comment (elewis787) [reply tests/unit_tests/db_engine_specs/test_crdb.py:36]: removing dttm causes the tests to fail. I can play around with this to see what the cause is if needed.

Comment (elewis787) [reply tests/unit_tests/db_engine_specs/test_crdb.py:36]: I used https://github.com/apache/superset/blob/master/tests/unit_tests/db_engine_specs/test_athena.py#L41 as a reference when writing the test.

**Comment by elewis787** [comment]
Any updates on this ?

**Comment by sfirke** [comment]
Not sure what's going on with CI here, I will close & reopen to try to trigger it

**Comment by sfirke** [comment]
@john-bodley and @eschutho can one of you peek at this and merge if OK?  Looks like the feedback has been addressed and CI has passed now that I kicked it (it had stalled 3 weeks ago).

**Review by john-bodley — APPROVED** [review]

**Pull request merged**